tools/makedeb

```diff
@@ -0,0 +1,267 @@
+#
+# Prerequisites:
+#     zsh, build-essential, devscripts, debhelper (>= 9)
+#
+# Reference: https://wiki.debian.org/IntroDebianPackaging.
+
+setopt errexit noshwordsplit nobashrematch
+[[ -n $DEBUG ]] && setopt xtrace
+
+################### SET UP ENVIRONMENT AND BASE DIRECTORIES ####################
+
+[[ -z $DEBFULLNAME ]] && DEBFULLNAME='Arun Prakash Jana'
+[[ -z $DEBEMAIL ]] && DEBEMAIL='arun@tuxdiary.com'
+[[ -z $TZ ]] && TZ='Asia/Kolkata'
+export DEBFULLNAME DEBEMAIL TZ
+
+here=$0:A:h
+repodir=$here/..
+builddir=$here/../build
+distdir=$here/../dist
+
+export GIT_DIR=$repodir/.git
+
+############################### HELPER FUNCTIONS ###############################
+
+print_error () print -R $'\e[31m'"Error: $*"$'\e[0m' >&2
+
+print_warning () print -R $'\e[33m'"Warning: $*"$'\e[0m' >&2
+
+# Usage: apt_package_version <package_name>
+apt_package_version () {
+    local version="${$(apt-cache show $1 | grep '^Version')#Version: }" || {
+        print_error "Version info not available for package ${(q-)1}."
+        exit 1
+    }
+    printf %s $version
+}
+
+debian_policy_version () {
+    local full_version="$(apt_package_version debian-policy)"
+    local match
+    [[ $full_version =~ ^(([0-9]+\.){2}[0-9]+) ]] || {
+        print_error "Invalid debian-policy version ${(q-)full_version}."
+        exit 1
+    }
+    printf %s $match[1]
+}
+
+# Git helpers
+
+# Usage: git_normalize_commitish <commit-ish>
+#
+# Normalize a commit-ish to a tag name if the commit-ish refers to a tag or
+# tagged commit; otherwise to a SHA (via git-rev-parse).
+git_normalize_commitish () {
+    local tagname commitsha
+    if tagname=$(git describe --exact --tags $1 2>/dev/null); then
+        printf %s $tagname
+    else
+        commitsha=$(git rev-parse --verify --quite $1) || {
+            print_error "Unable to parse ${(q-)1} as a git commit."
+            exit 1
+        }
+        printf %s $commitsha
+    fi
+}
+
+# Usage: git_ref_is_tag <refname>
+#
+# Returns 0 (is tag) or 1 (not tag).
+git_ref_is_tag () {
+    git show-ref --quiet --verify refs/tags/$1
+}
+
+# Usage: git_commitish_timestamp <normalized_commitish>
+#
+# Expects a normalized commit-ish (see git_normalize_commitish) as input, and
+# outputs a timestamp. For a tag, the timestamp is the tagger date. For a
+# commit, the timestamp is the committer date.
+git_commitish_timestamp () {
+    if git_ref_is_tag $1; then
+        local date="$(git for-each-ref --format='%(taggerdate)' refs/tags/$1)"
+
+        # The date returned by git-for-each-ref looks like `Sat Apr 23 10:38:27
+        # 2016 +0530', which isn't recognized by date(1) from coreutils. Need a
+        # hack to turn `+0530' into `U+0530' (same for -).
+        #
+        # Note that date will be empty if the tag at question is a lightweight
+        # tag (i.e., non-annotated) because they don't carry any tagger
+        # information; it that case we simply fall back to the committer.
```

Comment:
Aww hell, typo... Feels really bad when I need to rebase for a typo... I'll push a fixup commit for now and only squash after review.